13.2.2+: RenderSystem_GL linking with "-Wl,--no-undefined" broken

#### System Information
- Ogre Version: 13.2.4
- Operating System / Platform: openSUSE Tumbleweed (Linux)
- RenderSystem: :grey_question:

#### Detailled description
Compiling 13.1.1 works fine, but 13.2 is broken on openSUSE as of `undefined reference` error within the GLRenderSystem.
I attached a full build log, including cmake configuration and all packages that are installed.

#### Log
One example error, full log is [here build.log](https://github.com/OGRECave/ogre/files/7710701/build.log)

```
[  198s] /home/abuild/rpmbuild/BUILD/ogre-13.2.4/RenderSystems/GL/src/OgreGLRenderSystem.cpp: In member function '_setCullingMode':
[  198s] /home/abuild/rpmbuild/BUILD/ogre-13.2.4/RenderSystems/GL/src/OgreGLRenderSystem.cpp:1780:40: warning: 'cullMode' may be used uninitialized in this function [-Wmaybe-uninitialized]
[  198s]  1780 |         mStateCacheManager->setCullFace( cullMode );
[  198s]       |                                        ^
[  198s] /usr/lib64/gcc/x86_64-suse-linux/11/../../../../x86_64-suse-linux/bin/ld: /tmp/ccLEkj2R.ltrans0.ltrans.o: in function `PS_1_4::bindMachineInstInPassToFragmentShader(std::vector<unsigned int, std::allocator<unsigned int> > const&)':
[  198s] <artificial>:(.text+0xe8e): undefined reference to `glGetError'
[  198s] /usr/lib64/gcc/x86_64-suse-linux/11/../../../../x86_64-suse-linux/bin/ld: <artificial>:(.text+0x1139): undefined reference to `glGetError'
[  198s] /usr/lib64/gcc/x86_64-suse-linux/11/../../../../x86_64-suse-linux/bin/ld: /tmp/ccLEkj2R.ltrans0.ltrans.o: in function `Ogre::ATI_FS_GLGpuProgram::bindProgram()':
[  198s] <artificial>:(.text+0x24fb): undefined reference to `glEnable'
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,6 +32,10 @@ if(WIN32)
   set(CMAKE_SUPPRESS_REGENERATION true)
 endif()
 
+# prefer GLVND
+if(POLICY CMP0072)
+  cmake_policy(SET CMP0072 NEW)
+endif()
 # use legacy swig mode
 if(POLICY CMP0078)
   cmake_policy(SET CMP0078 OLD)

--- a/PlugIns/GLSLang/CMakeLists.txt
+++ b/PlugIns/GLSLang/CMakeLists.txt
@@ -17,7 +17,7 @@ elseif(WIN32)
     target_link_directories(Plugin_GLSLangProgramManager PRIVATE $ENV{VULKAN_SDK}/lib)
     target_link_libraries(Plugin_GLSLangProgramManager PUBLIC OgreMain shaderc_combined)
 else()
-    target_link_libraries(Plugin_GLSLangProgramManager PUBLIC OgreMain glslang HLSL OSDependent OGLCompiler SPIRV)
+    target_link_libraries(Plugin_GLSLangProgramManager PUBLIC OgreMain glslang HLSL OSDependent OGLCompiler SPIRV SPIRV-Tools SPIRV-Tools-opt)
 endif()
 
 ogre_config_framework(Plugin_GLSLangProgramManager)

--- a/RenderSystems/GL/CMakeLists.txt
+++ b/RenderSystems/GL/CMakeLists.txt
@@ -63,7 +63,7 @@ add_definitions(-DOGRE_GLPLUGIN_EXPORTS ${OGRE_VISIBILITY_FLAGS})
 #Note that in the next row SOURCE_FILES are added last. This is to prevent compilation problems of unity build found on Windows Visual Studio. 
 #In this situation any file added after the "glew.cpp" file, which belongs to the SOURCE_FILES package, does not compile
 add_library(RenderSystem_GL ${OGRE_LIB_TYPE} ${HEADER_FILES} ${GLSL_SOURCE} ${ATIFS_SOURCE} ${NVPARSE_SOURCE} ${SOURCE_FILES})
-target_link_libraries(RenderSystem_GL PUBLIC OgreMain PRIVATE OgreGLSupport)
+target_link_libraries(RenderSystem_GL PUBLIC OgreMain PRIVATE OgreGLSupport OpenGL::GL)
 target_include_directories(RenderSystem_GL PUBLIC 
     "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
     $<INSTALL_INTERFACE:include/OGRE/RenderSystems/GL>)

--- a/RenderSystems/Vulkan/CMakeLists.txt
+++ b/RenderSystems/Vulkan/CMakeLists.txt
@@ -29,9 +29,10 @@ elseif(ANDROID)
 else()
     set_source_files_properties("${CMAKE_CURRENT_SOURCE_DIR}/src/volk.c"
         PROPERTIES COMPILE_DEFINITIONS VK_USE_PLATFORM_XLIB_KHR)
+    target_link_libraries(RenderSystem_Vulkan PRIVATE ${X11_LIBRARIES})
 endif()
 
-target_link_libraries(RenderSystem_Vulkan PUBLIC OgreMain)
+target_link_libraries(RenderSystem_Vulkan PUBLIC OgreMain PRIVATE ${CMAKE_DL_LIBS})
 
 target_include_directories(RenderSystem_Vulkan PUBLIC
     "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"

--- a/ci-build.cmake
+++ b/ci-build.cmake
@@ -1,5 +1,5 @@
 set(GENERATOR)
-set(OTHER -DCMAKE_CXX_FLAGS=-Werror)
+set(OTHER -DCMAKE_CXX_FLAGS=-Werror -DCMAKE_SHARED_LINKER_FLAGS=-Wl,--no-undefined)
 set(CROSS)
 
 set(CMAKE_BUILD_TYPE Debug)

